tests/e2e/support/objects/app-files/share/actions.ts

```diff
@@ -60,21 +66,70 @@ export const createShare = async (args: createShareArgs): Promise<void> => {
 
 /**/
 
-export type ShareStatusArgs = Omit<ShareArgs, 'recipients'>
+export interface ShareStatusArgs extends Omit<ShareArgs, 'recipients'> {
+  via?: 'STATUS' | 'CONTEXT_MENU'
+}
 
 export const acceptShare = async (args: ShareStatusArgs): Promise<void> => {
-  const { resource, page } = args
-  await Promise.all([
-    page.locator(util.format(shareAcceptDeclineButton, resource, 'status-accept')).click(),
-    page.waitForResponse((resp) => resp.url().includes('shares') && resp.status() === 200),
-    page.locator(util.format(filesSharedWithMeAccepted, resource)).waitFor()
-  ])
+  const { resource, via, page } = args
+  if (via === 'CONTEXT_MENU') {
+    await clickActionInContextMenu({ page, resource }, 'accept-share')
+  } else {
+    await page.locator(util.format(shareAcceptDeclineButton, resource, 'status-accept')).click()
+    await page.waitForResponse(
+      (resp) =>
+        resp.url().includes('shares') && resp.status() === 200 && resp.request().method() == 'POST'
+    )
+  }
+  await page.locator(util.format(filesSharedWithMeAccepted, resource)).waitFor()
 }
 
 export const declineShare = async (args: ShareStatusArgs): Promise<void> => {
+  const { page, resource, via } = args
+  if (via === 'CONTEXT_MENU') {
+    await clickActionInContextMenu({ page, resource }, 'decline-share')
+  } else {
+    await page.locator(util.format(shareAcceptDeclineButton, resource, 'decline')).click()
+    await page.waitForResponse(
+      (resp) =>
+        resp.url().includes('shares') &&
+        resp.status() === 200 &&
+        resp.request().method() == 'DELETE'
+    )
+  }
+  await page.locator(util.format(filesSharedWithMeDeclined, resource)).waitFor()
+}
+
+export const clickActionInContextMenu = async (
+  args: ShareStatusArgs,
+  action: string
+): Promise<void> => {
   const { page, resource } = args
-  await page.locator(util.format(shareAcceptDeclineButton, resource, 'decline')).click()
-  await page.waitForResponse((resp) => resp.url().includes('shares') && resp.status() === 200)
+  let method = 'GET'
+  switch (action) {
+    case 'accept-share':
+    case 'create-quicklink':
+      method = 'POST'
+      break
+    case 'decline-share':
+      method = 'DELETE'
+      break
+  }
+
+  await Promise.all([
+    page.locator(util.format(actionMenuDropdownButton, resource)).click(),
+    page.waitForResponse(
+      (resp) =>
+        resp.url().includes('shared_with_me') &&
+        resp.status() === 200 &&
+        resp.request().method() === 'GET'
+    ),
+    page.locator(util.format(actionsTriggerButton, resource, action)).click(),
+    page.waitForResponse(
+      (resp) =>
+        resp.url().includes('shares') && resp.status() === 200 && resp.request().method() === method
+    )
```

Comment:
Does this happen sequentially or? Do the second click needs the first click to be completed?

Reply:
The `first click` is to open the context menu and the `second click` is to click action in the context menu. So those clicks must occur sequentially.

Reply:
then, is it okay to group them in a same promise resolver. They may run synchronously.

Reply:
This might become a problem in the future. The pattern should be
```js
await Promise.all([
  page.waitForResponse(...)
  page.<action>()
])
```
And always better to use only one action and one waitForResponse in one Promise.all block

Reply:
Thank you. :+1: